src/pykeen/nn/emb.py

```diff
@@ -1292,7 +1292,22 @@ def forward(
 
 
 class LabelBasedTransformerRepresentation(RepresentationModule):
-    """Label-based representations using a transformer encoder."""
+    """
+    Label-based representations using a transformer encoder.
+    
+    Example Usage:
+    
+    .. code-block ::
+        dataset = get_dataset(dataset="nations")
+        entity_representations = LabelBasedTransformerRepresentation.from_triples_factory(
+            triples_factory=dataset.training,
+        )
+        model = ERModel(
+            interaction="ermlp",
+            entity_representations=entity_representations,
+            relation_representations=EmbeddingSpecification(shape=entity_representations.shape),
+        )
```

Comment:
so in this formulation, the labels are initialized using the results from the transformer encoder? then stuff just continues on like normal?

Reply:
In this case, the Transformer model becomes part of the KGE, which is trained with the rest.

Reply:
The initializer does what you described before

Reply:
I added both descriptions to the docstrings in 
4488acb  and 
330a8f8

Reply:
can you give a bit more insight into why you'd want to do this? like is the philosophy to take a pre-trained transformer and fine tune it with a knowledge graph embedding model? and after this happens, you actually don't get any new embeddings besides this, right?

also can you add to the demo how to use the transformer inside the `LabelBasedTransformerRepresentation` after it's been trained?

Reply:
thanks for bearing with me. I'm sure the additions will make this much more useful (especally for pointing users to it in the future ;))

Reply:
First, labels are (often) a very strong signal, in particular when combined with an universal sentence encoder like BERT: Since these language models are trained on huge corpora, they learn meaningful representations, where vector similarity expresses semantic similarity. Thus, semantically similar entities (or more precisely, entities whose labels are semantically similar) get initialized with similar vectors, which may help the model to faster learn good representations for KGE.

If we use these representations for initialization, but do not keep the label encoder, we can think of this as a strong initialization.

However, if we keep the label -> vector mechanism part of the model during the training, we further make sure that even during training, labels with similar components continue to get mapped close by. In addition, we get (to some extent) out-of-the-box inductive capabilities.

As an example, consider the following model with label-based representations trained on nations
```python
from pykeen.pipeline import pipeline
from pykeen.datasets import get_dataset
from pykeen.nn.emb import EmbeddingSpecification, LabelBasedTransformerRepresentation
from pykeen.models import ERModel

dataset = get_dataset(dataset="nations")
entity_representations = LabelBasedTransformerRepresentation.from_triples_factory(
    triples_factory=dataset.training,
)
result = pipeline(
    dataset=dataset,
    model=ERModel,
    model_kwargs=dict(
        interaction="ermlpe",
        interaction_kwargs=dict(
            embedding_dim=entity_representations.embedding_dim,
        ),
        entity_representations=entity_representations,
        relation_representations=EmbeddingSpecification(
            shape=entity_representations.shape,
        ),
    ),
    training_kwargs=dict(
        num_epochs=1,
    ),
)
```

We can use the label-encoder part to generate representations for unknown entities labels. For instance, `"uk"` is an entity in `nations`, but we can also put in `"united kingdom"`, and get a roughly equivalent vector representations
```python
uk, united_kingdom = result.model.entity_representations[0].encoder(labels=["uk", "united kingdom"])
```

Thus, if we would put the resulting representations into the interaction function, we would get similar scores
```python
# true triple from train: ['brazil', 'exports3', 'uk']
scores = result.model.interaction(
    h=result.model.entity_representations[0].get_in_more_canonical_shape(
        dim="h",
        indices=torch.as_tensor(dataset.entity_to_id["brazil"]).view(1),
    ),
    r=result.model.relation_representations[0].get_in_more_canonical_shape(
        dim="r",
        indices=torch.as_tensor(dataset.relation_to_id["exports3"]).view(1),
    ),
    t=torch.stack([uk, united_kingdom]),
)
print(scores)
```

```

Reply:
Can you please find a place in the docs for that explanation?

Reply:
087d9bf 

@cthoyt feel free to iterate / move it to a better place